src/Symfony/Bridge/Doctrine/Tests/ContainerAwareEventManagerTest.php

```diff
@@ -38,19 +38,67 @@ public function testDispatchEvent()
         $this->assertTrue($listener2->called);
     }
 
+    public function testAddEventListenerAfterDispatchEvent()
+    {
+        $this->container->set('lazy1', $listener1 = new MyListener());
+        $this->evm->addEventListener('foo', 'lazy1');
+        $this->evm->addEventListener('foo', $listener2 = new MyListener());
+
+        $this->evm->dispatchEvent('foo');
+
+        $this->container->set('lazy2', $listener3 = new MyListener());
+        $this->evm->addEventListener('foo', 'lazy2');
+        $this->evm->addEventListener('foo', $listener4 = new MyListener());
+
+        $this->evm->dispatchEvent('foo');
+
+        $this->assertTrue($listener3->called);
```

Comment:
I think this should check if `$listener1` and `$listener2` were called during second dispatch